src/kubernetes/stream.rs

```diff
@@ -12,6 +12,8 @@ use snafu::{ResultExt, Snafu};
 
 /// Converts the HTTP response [`Body`] to a stream of parsed Kubernetes
 /// [`Response`]s.
+// Clippy rule, because `try_stream!`?
+#[allow(clippy::needless_return)]
```

Comment:
I was also able to resolve this before in #4818 when I mistakenly ran into it running the nightly toolchain by adding a unit return at the end of `try_stream!`: https://github.com/timberio/vector/pull/4818/files/471caf28164db3599611494d68efbcb0ab090ae6#diff-38f65b460ca0ce88f0e24ff9791321090396872183b32b03c8a3435dbe4f45f3

Reply:
Thanks @jszwedko! I rememer that I tried to fix this already but was not able issue/pr :disappointed: